synapse/storage/databases/main/devices.py

```diff
@@ -1079,6 +1102,8 @@ async def store_device(
             device_id: id of device
             initial_device_display_name: initial displayname of the device.
                 Ignored if device exists.
+            auth_provider_id: The SSO IdP the user used, if any.
+            oidc_sid: The session ID (sid) got from a OIDC login.
```

Comment:
What happens if these differ? (Can the same device end up logging it with a different OIDC session?)

Reply:
Good question. I wonder in what scenario this could happen though?
It would be more accurate if the oidc_sid & auth_provider_id were linked to the access token, but that means they have to be carried on when refreshing the access token, so I linked them to the device instead.

Reply:
Maybe if soft-logouts are enabled and the user has to login again on the same device? Would that trigger this?

Reply:
Do we have evidence that clients do that?
I see how it could happen with API calls, but not really in the wild with clients.
The other way around is more plausible (multiple devices/clients connected with the same OIDC session) and is supported by this